### PR TITLE
Pass the consul token as a header

### PR DESCRIPTION
**Pull request by akissa**
Headers are now the prefered way to pass the token to the consul
API - https://www.consul.io/api/index.html#authentication

### CODE REVIEW

File: patroni/dcs/consul.py [hunk 1]
```diff
@@ -82,6 +82,9 @@ def wrapper(callback, path, params=None, data=''):
                 kwargs['timeout'] = (float(params['wait'][:-1]) if 'wait' in params else 300) + 1
             else:
                 kwargs['timeout'] = self._read_timeout
+            if isinstance(params, dict) and 'token' in params and params['token']:
+                kwargs['headers'] = {'X-Consul-Token': params['token']}
```
Comment (CyberDem0n) [inline patroni/dcs/consul.py]: lines 85-86 could be rewritten as:
```python
kwargs['headers'] = {'X-Consul-Token': params.pop('token')}
```

My main concern was about breaking compatibility with the old Consul versions, but it seems that 'X-Consul-Token' support was added in version 0.6.0, released on December 3, 2015, so probably it should be fine.

Comment (akissa) [reply patroni/dcs/consul.py]: Using `.pop` could generate an attribute error, as it is possible for `params` to be `None`

Comment (CyberDem0n) [reply patroni/dcs/consul.py]: Previous line guaranties that 'token' is defined and even non empty

Comment (akissa) [reply patroni/dcs/consul.py]: Got you, so the change should be on line 86 then. Will update

**Comment by CyberDem0n** [comment]
:+1:

**Comment by outime** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
👍

**Comment by CyberDem0n** [comment]
Merged, thank you @akissa

**Pull request merged**